src/functions/module_lookup.ts

```diff
@@ -170,7 +170,12 @@ export const LOOKUP: AddFunctionDescription = {
       rangeLength,
       getElement
     );
-    assertAvailable(searchArray[0][index], searchKey);
+
+    if (resultRange === undefined && !verticalSearch && index !== -1) {
```

Comment:
mixin the `index === -1` case with this case makes it not particularly easy to read because it's technically not linked.
I would handle this case explicitly before. Something like this
```ts
if (index === -1) {
    assertAvailable(undefined, searchKey);
}
```